### ISSUE TITLE
sinatra layouts

### ISSUE THREAD

**Issue by croaky** [jferris/effigy#12]
Perhaps best in sinatra-effigy. What's the call on framework extensions existing inside or outside of Effigy? Saw another ticket about that.

**Comment by jferris** [comment]
Effigy is small enough that I think it's easiest to keep everything in one gem. If I get a patch that provides tested sinatra support, I'll apply it.

**Comment by croaky** [comment]
This is tested, but doesn't provide layout support. You want to bring in the Sinatra extension in some way first, then we can worry about layouts?

http://github.com/dancroak/sinatra-effigy

Not sure how'd you want to organize things. If you have an opinion, I can take a stab at it, too. May not pass Reek as-is.

**Comment by ghost** [comment]
I've been working on tilt-effigy for adding effigy support to tilt. I think it is going to be merged into tilt, and hopefully Sinatra itself eventually but I'm still waiting on a reply from Ryan Tomayko.

It supports:
- Layout & Partials
- Locals
- Access to the calling scope

http://github.com/robgleeson/tilt-effigy for anyone who is curious. The sinatra 1.0+ extension is available at http://github.com/robgleeson/sffigy and uses tilt-effigy behind the scenes.

Dan Croaks work is good too, but I found myself really wanting to have layouts with partials.

Thanks,
Rob
